step-framework-parent/step-framework-collections/src/main/java/step/core/ql/OQLAttributesAwareFilterVisitor.java

```diff
@@ -0,0 +1,119 @@
+     * @param attributesTransformFunction
+     * @param ignoreAttributes
+     */
+    public OQLAttributesAwareFilterVisitor(Function<String, String> attributesTransformFunction, Collection<String> ignoreAttributes) {
+        this.attributesTransformFunction = attributesTransformFunction != null ? attributesTransformFunction : Function.identity();
+        this.ignoreAttributes = ignoreAttributes;
+    }
+
+    @Override
+    public Filter visitEqualityExpr(OQLParser.EqualityExprContext ctx) {
+        String text0 = transform(unescapeStringIfNecessary(ctx.expr(0).getText()));
+        String text1 = unescapeStringIfNecessary(ctx.expr(1).getText());
+        if (ignoreAttributes.contains(text0)) {
+            return Filters.empty();
+        }
+        attributes.add(text0);
+        return processEqExpr(ctx, text0, text1);
+    }
+
+    @Override
+    public Filter visitComparisonExpr(OQLParser.ComparisonExprContext ctx) {
+        String text0 = transform(unescapeStringIfNecessary(ctx.expr(0).getText()));
+        if (ignoreAttributes.contains(text0)) {
+            return Filters.empty();
+        }
+        attributes.add(text0);
+        String text1 = unescapeStringIfNecessary(ctx.expr(1).getText());
+        return processComparisonExp(ctx, text0, text1);
+    }
+
+    protected String unescapeStringIfNecessary(String text1) {
+        if(text1.startsWith("\"") && text1.endsWith("\"")) {
+            text1 = unescapeStringAtom(text1);
+        }
+        return text1;
+    }
+
+    @Override
+    public Filter visitOrExpr(OQLParser.OrExprContext ctx) {
+        final Filter left = this.visit(ctx.expr(0));
+        final Filter right = this.visit(ctx.expr(1));
+        return Filters.or(List.of(left, right));
+    }
+
+    @Override
+    public Filter visitInExpr(OQLParser.InExprContext ctx) {
+        String text0 = transform(unescapeStringIfNecessary(ctx.expr().getText()));
+        if (ignoreAttributes.contains(text0)) {
+            return Filters.empty();
+        }
+        attributes.add(text0);
+        return processInExpr(ctx, text0);
+    }
+
+    @Override
+    public Filter visitNotExpr(OQLParser.NotExprContext ctx) {
+        final Filter expr = this.visit(ctx.expr());
+        return Filters.not(expr);
+    }
+
+    @Override
+    public Filter visitParExpr(OQLParser.ParExprContext ctx) {
+        return this.visit(ctx.expr());
+    }
+
+    @Override
+    public Filter visitNonQuotedStringAtom(OQLParser.NonQuotedStringAtomContext ctx) {
+        return Filters.fulltext(ctx.getText());
+    }
+
+    @Override
+    public Filter visitStringAtom(OQLParser.StringAtomContext ctx) {
+        String str = unescapeStringAtom(ctx.getText());
+        return Filters.fulltext(str);
+    }
+
+    protected String unescapeStringAtom(String str) {
+        // strip quotes
+        str = str.substring(1, str.length() - 1).replace("\"\"", "\"");
+        return str;
+    }
+
```

Comment:
@lucianbaciu Identical as super methods. No need to override. Use super methods instead